src/umap/umap.cpp

```diff
@@ -107,7 +117,8 @@ umapcfg_get_max_fault_events( void )
 }
 
 namespace Umap {
-
+  std::mutex m_mutex;
```

Comment:
These are global variables to the entire handler. They should be named in such a way that it is clear they are global. m_mutex doesn't convey that. Also num_threads. And there are no comments in the code to indicate that they are handler-wide global variables and why they are used.